sql: skip test with race

madelynnblue opened the pull request:
This was previously skipped, but the skip was removed last commit
because the linked issue in lib/pq was fixed. But there are some more
underlying races, so we must ignore it again.

Closes #12934

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/12951)
<!-- Reviewable:end -->

tamird commented:
Can you link to an upstream issue or issues? Both the commit message and
the Skip call, please.

On Tue, Jan 17, 2017 at 1:40 PM, Matt Jibson <notifications@github.com>
wrote:

> This was previously skipped, but the skip was removed last commit
> because the linked issue in lib/pq was fixed. But there are some more
> underlying races, so we must ignore it again.
>
> Closes #12934 <https://github.com/cockroachdb/cockroach/issues/12934>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/cockroachdb/cockroach/pull/12951
> Commit Summary
>
>    - sql: skip test with race
>
> File Changes
>
>    - *M* pkg/sql/copy_in_test.go
>    <https://github.com/cockroachdb/cockroach/pull/12951/files#diff-0> (2)
>
> Patch Links:
>
>    - https://github.com/cockroachdb/cockroach/pull/12951.patch
>    - https://github.com/cockroachdb/cockroach/pull/12951.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/12951>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPDibAa8o43Jh6WV5DxkstpDQX3MIks5rTQsagaJpZM4LmA1U>
> .
>

mjibson commented:
There isn't one yet. But I'm working on getting a repo in a lib/pq test and getting their whole test suite to be run with -race.

a-robinson reviewed (approved):
LGTM, but it'd be preferable to wait until there's an issue to point to before merging.

mjibson commented:
Agreed. I will modify the skip and commit messages to point to it before merging.

madelynnblue committed:
sql: skip test with race

This was previously skipped, but the skip was removed last commit
because the linked issue in lib/pq was fixed. But there are some more
underlying races, so we must ignore it again.

See: https://github.com/lib/pq/issues/558

Closes #12934

tamird commented:
Upstream issue is fixed, time to unskip these?

mjibson commented:
Sadly no, these tests uncover even more race conditions not covered by the lib/pq test suite where the underlying conn is accessed in parallel. I need to open an issue and look into it more.
